stl/inc/algorithm

```diff
@@ -10394,12 +10403,28 @@ namespace ranges {
 
             using _Memcmp_classification_pred =
                 typename decltype(_Lex_compare_memcmp_classify(_First1, _First2, _Pred))::_Pred;
-            if constexpr (!is_void_v<_Memcmp_classification_pred> && sized_sentinel_for<_Se1, _It1> //
-                          && sized_sentinel_for<_Se2, _It2> && same_as<_Pj1, identity> && same_as<_Pj2, identity>) {
+            constexpr bool _Is_sized1 = sized_sentinel_for<_Se1, _It1>;
+            constexpr bool _Is_sized2 = sized_sentinel_for<_Se2, _It2>;
+            if constexpr (!is_void_v<_Memcmp_classification_pred> //
+                          && (_Is_sized1 || same_as<_Se1, unreachable_sentinel_t>) //
+#pragma warning(suppress : 6287) // Redundant code: the left and right subexpressions are identical
+                          &&(_Is_sized2 || same_as<_Se2, unreachable_sentinel_t>) //
+                          &&same_as<_Pj1, identity> && same_as<_Pj2, identity>) {
                 if (!_STD is_constant_evaluated()) {
-                    const auto _Num1 = static_cast<size_t>(_Last1 - _First1);
-                    const auto _Num2 = static_cast<size_t>(_Last2 - _First2);
-                    const int _Ans   = _Memcmp_count(_First1, _First2, (_STD min)(_Num1, _Num2));
+                    size_t _Num1, _Num2;

```

Comment:
We conventionally avoid declaring multiple variable in a single line. Please move `_Num2` below so